overlay.nix

```diff
@@ -0,0 +1,28 @@
+let
```

Comment:
It seems like this refactor is not 100% necessary, but I'm not opposed.

Reply:
Definitely not necessary, but otherwise there were multiple levels of let-in, which became confusing to read.